include/swift/AST/ASTMangler.h

```diff
@@ -510,10 +512,22 @@ class ASTMangler : public Mangler {
   /// \param contextSig The signature of the known context. This function
   /// will only mangle the difference between \c sig and \c contextSig.
   ///
+  /// \param skipEquivalenceCheck Whether we should skip the sig == contextSig
```

Comment:
Please remove this

Reply:
There has to be a simpler way to solve this problem. I'm having trouble following the logic in all of the changes in this PR. Please rethink the approach.